### DISCUSSION TITLE
Remove hard Trix dependency?

### DISCUSSION THREAD

**Discussion by evenreven** [SpinaCMS/Spina#1067] [Ideas]
I think Spina is a really exciting project that I want to try on a project, but the dealbreaker for me is Trix. It's just a very bad editor, in my opinion, both for the lack of customisation and the terrible markup it provides. 

Would you consider - at some point in the future - making the Trix integration a separate gem and let the community create their own integrations (with quill, tiptap, tiny or whatever)? This is the route RefineryCMS took, they decoupled the wymeditor integration from Refinery Core and made it a gem instead so the community could take care of the rest. 

Or does Action Text and its tight coupling with Trix make this impossible?

**Comment by Bramjetten** [comment]
Spina doesn’t use ActionText, Spina uses a custom implementation of Trix just for the Spina::Parts::Text model. There’s no tight coupling, it’s easy to support a different editor by adding a new Spina part (i.e. Spina::Parts::Quill). 

We welcome new gems that add support for other editors!

For now, Trix will remain Spina’s default which we’ll include in the gem. But if something better comes along we’d be happy to change that default.

**Reply by evenreven** [reply]
Oh, interesting! Thank you so much for the quick response. I'll have a look at the feasibility of that approach. 👍
